packages/common-integration-tests/src/leaderboard-client.ts

```diff
@@ -0,0 +1,1072 @@
+        integrationTestCacheName,
+        leaderboardName,
+        {
+          minScore: 100,
+          maxScore: 50,
+        }
+      );
+      expectWithMessage(() => {
+        expect(badRange).toBeInstanceOf(LeaderboardFetch.Error);
+      }, `expected ERROR but got ${badRange.toString()}`);
+      const badRangeError = badRange as LeaderboardFetch.Error;
+      expectWithMessage(() => {
+        expect(badRangeError.errorCode()).toEqual(
+          MomentoErrorCode.INVALID_ARGUMENT_ERROR
+        );
+      }, `expected INVALID_ARGUMENT_ERROR but got ${badRangeError.errorCode()} ${badRangeError.message()}`);
+
+      // undefined min and max are acceptable
+      const undefinedRange = await leaderboardClient.leaderboardFetchByScore(
+        integrationTestCacheName,
+        leaderboardName
+      );
+      expectWithMessage(() => {
+        expect(undefinedRange).toBeInstanceOf(LeaderboardFetch.Found);
+      }, `expected Found but got ${undefinedRange.toString()}`);
+      const receivedElements = (
+        undefinedRange as LeaderboardFetch.Found
+      ).valueArray();
+      expect(receivedElements).toEqual([]);
+    });
+
+    it('fetches elements given a variety of score ranges', async () => {
+      // Insert some elements
+      const elements = new Map([
+        [BigInt(123), 10.0],
+        [BigInt(234), 100.0],
+        [BigInt(345), 250.0],
+        [BigInt(456), 500.0],
+        [BigInt(567), 750.0],
+        [BigInt(678), 1000.0],
+        [BigInt(789), 1500.0],
+        [BigInt(890), 2000.0],
+      ]);
+      const response = await leaderboardClient.leaderboardUpsert(
+        integrationTestCacheName,
+        leaderboardName,
+        elements
+      );
+      expectWithMessage(() => {
+        expect(response).toBeInstanceOf(LeaderboardUpsert.Success);
+      }, `expected SUCCESS but got ${response.toString()}`);
+
+      // Fetch using offset and count: failing, returning nothing
+      // const fetchWithOffsetAndCount =
+      //   await leaderboardClient.leaderboardFetchByScore(
+      //     integrationTestCacheName,
+      //     leaderboardName,
+      //     {
+      //       offset: 2,
+      //       count: 2,
+      //     }
+      //   );
+      // expectWithMessage(() => {
+      //   expect(fetchWithOffsetAndCount).toBeInstanceOf(LeaderboardFetch.Found);
+      // }, `expected HIT but got ${fetchWithOffsetAndCount.toString()}`);
+      // const receivedWithOffsetAndCount = (
+      //   fetchWithOffsetAndCount as LeaderboardFetch.Found
+      // ).valueArray();
+      // const expectedWithOffsetAndCount = [
+      //   {
+      //     id: BigInt(345),
+      //     score: 250.0,
+      //     rank: BigInt(2),
+      //   },
+      //   {
+      //     id: BigInt(456),
+      //     score: 500.0,
+      //     rank: BigInt(3),
+      //   },
+      // ];
+      // expect(receivedWithOffsetAndCount).toEqual(expectedWithOffsetAndCount);
```

Comment:
@tylerburdsall @virratanasangpunth 
Thanks for addressing the GetRank bugs!

This GetByScore test is still failing though. Are unbounded min and max implemented for this API? 
Tried using Postman to make the same request and I also received an empty array when I was expecting to receive 2 elements.

Reply:
> Are unbounded min and max implemented for this API?

yes, this is implemented. If you set the offset to `0` and the `count` to something like `99`, what are the results?

Reply:
I still get `"elements": []`

Reply:
Hmm that's odd, by default we should be handing this. But you're right @anitarua, something is happening here and I reproduced this locally. I will dig into this

Reply:
Ah, found the source of the bug! I'll start pushing the fix on our back end

Reply:
This should be fixed now!